lexicon/providers/ovh.py

```diff
@@ -0,0 +1,189 @@
+    'soyoustart-ca': 'https://ca.api.soyoustart.com/1.0',
+}
+
+def ProviderParser(subparser):
+    subparser.description = '''
+        OVH Provider requires a token with full rights on /domain/*.
+        It can be generated for your OVH account on the following URL: 
+        https://api.ovh.com/createToken/index.cgi?GET=/domain/*&PUT=/domain/*&POST=/domain/*&DELETE=/domain/*'''
+    subparser.add_argument('--auth-entrypoint', help='specify the OVH entrypoint', choices=[
+        'ovh-eu', 'ovh-ca', 'soyoustart-eu', 'soyoustart-ca', 'kimsufi-eu', 'kimsufi-ca'
+        ])
+    subparser.add_argument('--auth-application-key', help='specify the application key')
+    subparser.add_argument('--auth-application-secret', help='specify the application secret')
+    subparser.add_argument('--auth-consumer-key', help='specify the consumer key')
+
+class Provider(BaseProvider):
+
+    def __init__(self, options, engine_overrides=None):
+        super(Provider, self).__init__(options, engine_overrides)
+
+        # Handling missing required parameters
+        if not self.options.get('auth_entrypoint'):
+            raise Exception('Error, entrypoint is not defined')
+        if not self.options.get('auth_application_key'):
+            raise Exception('Error, application key is not defined')
+        if not self.options.get('auth_application_secret'):
+            raise Exception('Error, application secret is not defined')
+        if not self.options.get('auth_consumer_key'):
+            raise Exception('Error, consumer key is not defined')
+
+        # Construct DNS OVH environment
+        self.domain_id = None
+        self.endpoint_api = ENDPOINTS.get(self.options.get('auth_entrypoint'))
+
+        # All requests will be done in one HTTPS session
+        self.session = requests.Session()
+
+        # Calculate delta time between local and OVH to avoid requests rejection
+        server_time = self.session.get('{0}/auth/time'.format(self.endpoint_api)).json()
+        self.time_delta = server_time - int(time.time())
+
+    def authenticate(self):
+        domain = self.options.get('domain')
+
+        domains = self._get('/domain/zone/')
+        if domain not in domains:
+            raise Exception('Domain {0} not found'.format(domain))
+
+        status = self._get('/domain/zone/{0}/status'.format(domain))
+        if not status['isDeployed']:
+            raise Exception('Zone {0} is not deployed'.format(domain))
+
+        self.domain_id = domain
+
+    def create_record(self, type, name, content):
+        domain = self.options.get('domain')
+        ttl = self.options.get('ttl')
+
+        data = {
+            'fieldType': type,
+            'subDomain': self._relative_name(name),
+            'target': content
+        }
+
+        if ttl:
+            data['ttl'] = ttl
+
+        self._post('/domain/zone/{0}/record'.format(domain), data)
+        self._post('/domain/zone/{0}/refresh'.format(domain))
+
```

Comment:
Add `logger.debug` message for all actions so that users calling Lexicon via the CLI get a status message. https://github.com/AnalogJ/lexicon/blob/master/lexicon/providers/cloudflare.py#L75